### ISSUE TITLE
Errors exporting model from PyTorch to Caffe2

### ISSUE THREAD

**Issue by peastman** [pytorch/pytorch#7650]
## System Info

PyTorch 0.4.0 (installed with conda)
Caffe2 0.8.dev (installed with conda)
onnx-caffe2 1.0.0 (installed with pip)
macOS 10.13.4
Python 3.6.5
no CUDA

## Issue description

I'm trying to build a model with PyTorch, export it as a Caffe2 model, then use it in a C++ program.  I'm pretty sure the C++ code is correct.  At any rate, it works correctly if I use a model built directly with Caffe2.  But I run into various problems when using a PyTorch model.  Here is the code I use to generate it:

```python
import torch
import torch.nn as nn

class Compute(nn.Module):
    def forward(self, x):
        return torch.sum(x**2)

x = torch.rand(10, 3)
torch.onnx.export(Compute(), x, "test.onnx", verbose=True, input_names=['positions'], output_names=['energy'])
```

I convert it to a Caffe2 model with `convert-onnx-to-caffe2`, then try to execute it in my C++ program.  It fails with this error:

```
exception: [enforce fail at tensor.h:495] IsType<T>(). Tensor type mismatch, caller expects elements to be float while tensor contains long long Error from operator: 
input: "positions" input: "1" output: "2" name: "" type: "Pow" device_option { device_type: 0 cuda_gpu_id: 0 }
** while accessing input: 1
```

The problem appears to be the operation `x**2`.  PyTorch is recording the exponent as being a `long long`, but Caffe2 insists it must be a `float`.

As a temporary workaround, I tried eliminating the power operation by changing the line to `return torch.sum(x*x)`.  With that change I can run the model, but when I query the "energy" output, it's wrong.  It ought to be a scalar containing the sum of squares of all the input elements.  Instead, it comes out as a (10, 3) matrix containing the square of each element.  That is, the sum operation is never getting run.

**Comment by bddppq** [comment]
@peastman Thanks for reporting! I have put a fix PR in #7657

**Comment by peastman** [comment]
Thanks!  It looks like that's a fix for the error in Pow?  What about the missing sum?  I assume that's a different problem?

**Comment by peastman** [comment]
I don't think this should have been closed yet?  The issue with the sum operation being skipped still hasn't been resolved.

**Comment by bddppq** [comment]
@peastman Ah sorry I missed the last paragraph of your description. I have put up another PR to fix that and added test cases for it. #7685

**Comment by peastman** [comment]
Perfect, thanks!

**Comment by bddppq** [comment]
@peastman The fix has been merged now, please give it a try and kindly close this issue after verifying. Thanks

**Comment by peastman** [comment]
Sorry for the delay on this.  I've been trying for several days to compile from source so I could test it, but I seem to be down a rabbit hole where everything I do breaks something else.  I first uninstalled the conda build of PyTorch, and after some effort (see #7764) managed to do that.  Then I uninstalled the conda build of Caffe2 (probably a mistake) and tried to compile a consistent build of that from the same repository revision.  That took quite a bit more effort (see #7770), but eventually I succeeded.  Only then I couldn't get `convert-onnx-to-caffe2` to work anymore.  Eventually I decided that compiling Caffe2 had been a mistake, so I did my best to uninstall it (difficult, because it spews files all over the filesystem) and reinstalled the conda build.  I was then able to convert my files.  But somewhere in all of that, something else in my system has gotten broken.  My C++ program that uses Caffe2 doesn't work anymore.  When I run it, I get this error message:

```
ERROR: flag 'help' was defined more than once (in files '/opt/concourse/worker/volumes/live/26a9f68c-5f43-4c42-5672-33448f6eb48d/volume/gflags_1512755328973/work/src/gflags_reporting.cc' and '/tmp/gflags-20170916-46768-ftgkio/gflags-2.2.1/src/gflags_reporting.cc').
```

Needless to say, neither one of the files it's complaining about a conflict between actually exists on my computer...
